linux/LibNl.c

```diff
@@ -75,6 +75,12 @@ static void unload_libnl(void) {
 }
 
 static int load_libnl(void) {
+
+   DECLARE_ELF_NOTE_DLOPEN("[{\"soname\":[\"libnl-3.so\",\"libnl-3.so.200\"],\"feature\":\"delay-accounting\"," \
+      "\"description:\":\"Enables delay accounting support\",\"priority\":\"recommended\"},{\"soname\":[" \
+      "\"libnl-genl-3.so\",\"libnl-genl-3.so.200\"],\"feature\":\"delay-accounting\",\"description:\":"\
+      "\"Enables delay accounting support\",\"priority\":\"recommended\"}]")
```

Comment:
Any objections to have this literal span multiple lines for better maintainability?

Please add proper meta-commands for `astyle` to leave this block alone afterwards.

Also, this note probably should only contain the optional dependencies we actually have enabled; i.e. if configured without delay accounting, there's no need to declare this as an optional runtime dependency if we would never try to load that library anyway.

```suggestion
   DECLARE_ELF_NOTE_DLOPEN( "["\
      "{"\
         "\"soname\":[\"libnl-3.so\",\"libnl-3.so.200\"],"\
         "\"feature\":\"delay-accounting\"," \
         "\"description:\":\"Enables delay accounting support\","\
         "\"priority\":\"recommended\""\
      "},"\
      "{"\
         "\"soname\":[\"libnl-genl-3.so\",\"libnl-genl-3.so.200\"],"\
         "\"feature\":\"delay-accounting\","\
         "\"description:\":\"Enables delay accounting support\","\
         "\"priority\":\"recommended\""\
      "}"\
   "]" )
```

Reply:
I'm not a fan of having a full JSON syntax in a macro caller. Can we rework the macros so that only important fields will be uses as parameters?

I expect the usage would ideally work like this:
```c
#define ELF_NOTE_DLOPEN_DEP(sonames, feature_tag, priority) ...
// Example
ELF_NOTE_DLOPEN_DEP(
   "[\"libnl-3.so\",\"libnl-3.so.200\"]", "delay-accounting", "recommended")
ELF_NOTE_DLOPEN_DEP(
   "[\"libnl-genl-3.so\",\"libnl-genl-3.so.200\"]", "delay-accounting", "recommended")
```

How do you think?

Reply:
While I also thought about this, including making generating the JSON be split into different macro calls (definitively possible AFAICS), there's one minor issue with the pre-processor IMHO, that prevents this from being really nice syntax. The issue is, that you can't properly auto-terminate lists of arguments to insert the semicolon in list contexts, thus you'd most likely have to resort to something like `"[" ELF_NOTE_DEP(foo…) "," ELF_NOTE_DEP(bar…) "]"`; which doesn't look too terrible, but in a really great world could work without the explicit "," token in between.

Also from a practcal PoV, I'd prefer such macros to go ELF_NOTE_DLOPEN_DEP(freature, level, description, solist)`, but comparibly that's a minor change to implement.